src/Sarif/Visitors/ReformattingVisitor.cs

```diff
@@ -43,23 +45,129 @@ public override Run VisitRun(Run node)
                 }
             }
 
+            if (node.Results != null)
+            {
+                var results = new List<Result>();
+
+                foreach (Result result in node.Results)
+                {
+                    results.Add(VisitResult(result));
+                }
+
+                node.Results = results;
+            }
+
             return node;
         }
 
-        internal FileData VisitDictionaryValueNullChecked(string key, FileData node)
+        public override PhysicalLocation VisitPhysicalLocation(PhysicalLocation node)
         {
-            Uri uri;
+            // If we have a binary region, there is no work to do
+            if (node.Region == null ||
+                (node.Region.ByteOffset > 0 || node.Region.ByteLength > 0))
+            {
+                goto Exit;
+            }
+
+            // If we already have a snippet, let's assume it is correct
+            if (node.Region.Snippet != null)
+            {
+                goto Exit;
+            }
 
-            if (!Uri.TryCreate(key, UriKind.RelativeOrAbsolute, out uri))
+            // If the user hasn't specified code snippet insertion, there is no work to do
+            bool workToDo = false;
+            workToDo |= node.Region != null && _dataToInsert.Includes(OptionallyEmittedData.CodeSnippets);
+
+            if (!workToDo)
+            {
+                goto Exit;
+            }
+
+            // If we have an absolute URL specified as a file that 
+            // exist on disk, there is work to do
+            if (node.FileLocation.Uri.IsAbsoluteUri &&
+                node.FileLocation.Uri.LocalPath != null &&
+                File.Exists(node.FileLocation.Uri.LocalPath))
```

Comment:
The SARIF SDK has an `IFileSystem` that you can inject to make this more easily unit testable.

Reply:
Resolved. Will expand unit tests in next check-in as well.

---
In reply to: [202102007](https://github.com/Microsoft/sarif-sdk/pull/942#discussion_r202102007) [](ancestors = 202102007)